pages/api/auth/login.ts

```diff
@@ -0,0 +1,13 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import nc from "next-connect";
+import mongoose from "../../../lib/middlewares/mongoose";
+import auth from "../../../lib/middlewares/auth";
+import passport from "../../../lib/auth/passport";
```

Comment:
Good idea to split all of this into multiple files to make it easier to read the distinct components. It might be worth adding a comment in some of these files to summarize what they do in the context of the larger OAuth process.